xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCUser.java

```diff
@@ -574,6 +575,36 @@ private void processDecliningInvitationMessage(
                                          info.elementTextTrim("reason"), packet.getFrom());
     }
 
+    /**
+    * checks for memberlist retrival query
+    * @return true if occupant is allowed to retrieve member list otherwise false
+    */
+    private boolean isAllowed(IQ packet)
```

Comment:
This is a confusing implementation, which would be very hard to maintain.

The method name is very generic - but is checks for a very specific thing. Reflecting that in the method name would be a good improvement.

The method checks if a particular entity has permission to do so. To check permission, it should not be needed to check what the content of the stanza is (if it is a muc#admin query). This probably does serve a purpose - but if a check like this is needed, then it should go in a dedicated method.